File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char, pair<bool,int>> mp;
        for(int i = 0; i < s.length(); i++){
            if(mp.find(s[i]) == mp.end()){
                mp.insert({s[i],{false, i}});
            }else{
                mp[s[i]] = {true, i};
            }
        }
        int ans = s.length();
        bool flag = false;
        for(auto iter : mp){
            if(iter.second.first == false && iter.second.second < ans){
                flag = true;
                ans = iter.second.second;
            }
        }
        return flag ? ans : -1;
    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cs
public class Solution {
    public int FirstUniqChar(string s) {
        Dictionary<char, int> dic = new Dictionary<char, int>();
        for(int i = 0; i < s.Length; i++){
            if(dic.ContainsKey(s[i])){
                dic[s[i]]++;
            }else{
                dic.Add(s[i], 1);
            }
        }
        for(int i = 0; i < s.Length; i++){
            if(dic[s[i]] == 1)
                return i;
        }
        return -1;
    }
}